lib/CArtHandler.h

```diff
@@ -386,7 +386,28 @@ namespace ArtifactUtils
 	DLL_LINKAGE ArtifactPosition getArtifactDstPosition(	const CArtifactInstance * artifact, 
 								const CArtifactSet * target,
 								ArtBearer::ArtBearer bearer);
-	DLL_LINKAGE std::vector<ArtifactPosition> unmovablePositions(); // TODO: Make this constexpr when the toolset is upgraded
+	const std::array<ArtifactPosition::EArtifactPosition, 2> unmovableSlots =
+	{
+		ArtifactPosition::SPELLBOOK,
+		ArtifactPosition::MACH4
+	};
+	const std::array<ArtifactPosition::EArtifactPosition, 14> constituentWornSlots =
+	{
+		ArtifactPosition::HEAD,
+		ArtifactPosition::SHOULDERS,
+		ArtifactPosition::NECK,
+		ArtifactPosition::RIGHT_HAND,
+		ArtifactPosition::LEFT_HAND,
+		ArtifactPosition::TORSO,
+		ArtifactPosition::RIGHT_RING,
+		ArtifactPosition::LEFT_RING,
+		ArtifactPosition::FEET,
+		ArtifactPosition::MISC1,
+		ArtifactPosition::MISC2,
+		ArtifactPosition::MISC3,
+		ArtifactPosition::MISC4,
+		ArtifactPosition::MISC5,
+	};

```

Comment:
Not sure what's better - this const array or function that return reference to const vector (with all details such as size & elements hidden inside)
But this is minor detail, looks OK othervice.

Reply:
I personally would prefer to have such a hardcoded arrays as it for now declared in header without hiding them inside function. But it's just my preference

Reply:
changing headers causes recompilation so I prefer keeping implementation details in cpp units as more as it is possible

Reply:
Also I am not sure if const variables should be uppercased or not

Reply:
Also this data may be moved to ArtifactUtils. near isBackpackSlot and as far as I understoot it is heroSlots - unmovable slots, right? So these are hero movable slots and such data may be usable for some other code.

Reply:
Or even put it into ArtifactPosition. ArtifactPosition::isBackpackSlot(pos), ArtifactPosition::movableHeroSlots(), ArtifactPosition::unmovableHeroSlots()

Reply:
> as far as I understoot it is heroSlots - unmovable slots, right?

This list also excludes all war machines, which are neither "unmovableSlots" nor "constituentWornSlots".
Kind of makes sense - while war machines may be *theoretically* used as constituent art for combined, this is not something that make sense right now and likely to cause issues.

Reply:
in that case - perhaps move it into .cpp? Something like this:
```
const std::vector<ArtifactPosition::EArtifactPosition> & constituentWornSlots()
{
    static const std::vector<ArtifactPosition::EArtifactPosition>slots = { <elements> }

    return slots;
}
```
This way all internal details like elements & their amount is hidden from the interface & function overhead is minimal since it just returns existing vector.

Would also help with reducing header size - less noise if you're looking through API.

Reply:
Ok, makes sense.

Reply:
> Kind of makes sense - while war machines may be _theoretically_ used as constituent art for combined, this is not something that make sense right now and likely to cause issues.

It was assumed that the list could be expanded in the future